### DISCUSSION TITLE
2021-25: mv 与 rename

### DISCUSSION THREAD

**Discussion by Xuanwo** [Xuanwo/blog#213] [Announcements]
https://xuanwo.io/reports/2021-25/ 

 An infrastructure engineer, focused on distributed storage system

**Comment by Prnyself** [comment]
这就是我和大佬的区别了，多年以前我在 Win 上剪切游戏存档的时候也考虑过这个问题，但是没有深究下去了

**Comment by gaocegege** [comment]
挺有意思的

**Comment by zdyxry** [comment]
看到了一个帖子： https://lore.kernel.org/linux-fsdevel/CAJCQCtQ38W2r7Cuu5ieKRQizeKF0tf--3Z8yOJeeR+ZZ4S6CVQ@mail.gmail.com/

> I can't tell from documentation if renameat2() with flag
RENAME_NOREPLACE is atomic, assuming the file doesn't exist at
destination.
